locust/shape.py

```diff
@@ -0,0 +1,35 @@
+import time
+
+
+class LoadTestShape(object):
+    """
+    A simple load test shape class used to control the shape of load generated
+    during a load test.
+    """
+
+    start_time = time.monotonic()
+
+    def reset_time(self):
+        """
+        Resets start time back to 0
+        """
+        self.start_time = time.monotonic()
+
+    def get_run_time(self):
+        """
+        Calculates run time in seconds of the load test
+        """
+        return time.monotonic() - self.start_time
+
+    def tick(self):
+        """
+        Returns a tuple with 3 elements to control the running load test.
+
+            user_count -- Total user count
+            hatch_rate -- Hatch rate to use when changing total user count
+            stop_test -- A boolean to stop the test
+
+        """
+        run_time = self.get_run_time()
+
+        return (0, 0, True)
```

Comment:
@cyberw comments about what to return should go here

Reply:
> I meant that tick() should do return None instead of return (0, 0, True)

Is that consistent? Wouldn't something like `(Int, Int)`, `(None, None)` or `()` be better?

Reply:
I dont know what you mean by (Int, Int) - how would that be used indicate the end of the test?

I think None is the most consistent of the options, also it requires the least amount of code and thought from the user. 

It is not a ramp down to null users using ramping down at a rate of null per second, so your second option doesnt make sense to me.

I guess we could do "anything that is "falsy" (to allow `()`, `None` and `False`), if that sounds better to you.

Reply:
I meant trying to keep the type that is returned consistent in all cases. It returns a tuple (of 2 integers normally `(Int, Int)`) so if you we want to be consistent then it should always return a tuple, perhaps sometimes an empty one, `()`.

But I know this is a [contentious topic in Python](https://www.reddit.com/r/Python/comments/30yb5t/return_none_or_not_to_return_none/), especially for people coming from other languages who find it a bit too loose.

Ultimately I don't care enough to write more comments about this so I'll just change it to return `(Int, Int)` or `None` 😃